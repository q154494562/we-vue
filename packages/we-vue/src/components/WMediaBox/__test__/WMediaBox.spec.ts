import { shallowMount } from '@vue/test-utils'
import MediaBox from '../'

describe('media-box', () => {
  test('create', () => {
    let wrapper = shallowMount(MediaBox, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-media-box')
    expect(wrapper.html()).toMatchSnapshot()

    wrapper = shallowMount(MediaBox, {
      propsData: {
        type: 'text',
      },
    })

    expect(wrapper.name()).toBe('w-media-box')
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('handle click', () => {
    const routeLinkSpy = jest.fn()
    const wrapper = shallowMount(MediaBox, {
      propsData: {},
      methods: {
        routeLink: routeLinkSpy,
      },
    })

    wrapper.trigger('click')
    expect(wrapper.emitted().click).toBeTruthy()
    expect(routeLinkSpy).toHaveBeenCalled()
  })
})
