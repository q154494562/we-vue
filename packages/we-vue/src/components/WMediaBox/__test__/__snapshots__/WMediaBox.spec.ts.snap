// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media-box create 1`] = `
<div class="weui-media-box weui-media-box_appmsg">
  <div class="weui-media-box__hd"><img class="weui-media-box__thumb"></div>
  <div class="weui-media-box__bd">
    <h4 class="weui-media-box__title"></h4>
    <p class="weui-media-box__desc"></p>
  </div>
</div>
`;

exports[`media-box create 2`] = `
<div class="weui-media-box weui-media-box_text">
  <h4 class="weui-media-box__title"></h4>
  <p class="weui-media-box__desc"></p>
</div>
`;
